torch/csrc/jit/codegen/cuda/tensor_view.cpp

```diff
@@ -423,6 +432,178 @@ TensorView* TensorView::computeAt(
   return this;
 }
 
+void TensorView::computeWith(int pos, bool best_effort) {
+  TORCH_INTERNAL_ASSERT(
+      !container()->isA<kir::Kernel>(),
+      "Function invalid for kernel container.");
+
+  if (isFusionInput()) {
+    return;
+  }
+
+  TORCH_CHECK(
+      !ir_utils::consumerTvsOf(this).empty(),
+      "There must be at least one consumer of this tensor to use computeWith: ",
+      toString());
+
+  if (pos < 0) {
+    pos += int(nDims()) + 1;
+  }
+
+  TORCH_INTERNAL_ASSERT(
+      pos >= 0 && pos <= nDims(),
+      "Invalid inline position for ",
+      toString(),
+      ": ",
+      pos);
+
+  const auto max_inline_pos =
+      MaxPosCalculator({}, true).getMaxPosAll(this, best_effort);
+
+  if (best_effort) {
+    pos = std::min<int>(max_inline_pos, pos);
+  }
+
+  // hoist inner most broadcast
+  while (pos > 0 && axis(pos - 1)->isBroadcast()) {
+    pos--;
+  }
+
+  TORCH_CHECK(
+      pos <= max_inline_pos,
+      "Invalid computeWith position for T",
+      name(),
+      ": ",
+      pos,
+      ". Maximum allowed value:",
+      max_inline_pos);
+
+  // The position must be right of the computeAt position
+  TORCH_CHECK(
+      pos >= getComputeAtPosition(),
+      "Position must be right of the computeAt position. Position: ",
+      pos,
+      ", computeAt position: ",
+      getComputeAtPosition());
+
+  // If it's already set to be computed with the consumer and the
+  // position is higher, nothing to change
+  if (getComputeWithPosition() >= pos) {
```

Comment:
Do we need to update consumers in some way in this call to make sure that consumers don't transform in an inconsistent way after the computeWith of their producers is set?

Reply:
It seems to me `resolveComputeWith` will simply fail if we're in an inconsistent state?

Reply:
Ah, I see. Since at this point no consumer is aware of the potential computeWith into it, there's nothing to block the consumers to be transformed in an inconsistent way. I need to think about this more carefully.

Reply:
I don't think this needs to be a blocker for this PR, we might just want to more explicitly validate in lowering.

Reply:
Would be nice in the future to error on transformation attempt of the consumer.

Reply:
Added `maybe_max_producer_pos_` (1c83e699fe495ec158772ee1e7eaddb8cbe443bc). Don't like the name, but don't have any other idea.

Modifying consumer domains where producer may be computed at should now throw an error.